docs/enhancements/add-feature-support-by-generic-attributes.md

```diff
@@ -0,0 +1,144 @@
+Whenever a user wants to know if a feature is supported in an OpenShift version, the user needs to call 
+the feature-support-list API, get the entire list and filter the results to find the specific feature for that version.
+As part supporting multiple CPU architectures on Assisted Installer, we want the ability to add another dimension to the 
+returned object - CPU architecture, so we can determine if a feature is supported or not by two parameters instead of one.
+
+
+## Motivation
+Adding another attribute to the feature-support-level API is indeed required but in its current implementation there are 
+few flaws (from my perspective):
+1. The current implementation forces us to duplicate the entire features list for each cpu architecture (API change).\
+For example if for `OpenshiftVersion = 4.12` we currently hold 10 features so adding just ARM64 as a CPU architecture to
+that equation will require us to duplicate the 10 features (for that architecture) and set the support-level for both 
+x86 and arm64.
+2. The list is statically defined in assisted-service code so on new OCP version we will have to duplicate the entire 
+features from the previous version even if it supported or not supported for few past versions.
+3. It is hard to expand, beside of adding CPU architecture to that equation if in the future we will want to filter by a
+third attribute we will have to duplicate everything again (API change).
+4. The user is have to get the entire feature-support list even if he needs to know just the support of a single feature.
+
+These four reasons led me to the realization that we must create a more flexible and easier-to-maintain API. 
+
+### Goals
+
+- Add another dimension (of CPU architecture) to the feature-supported endpoint  
+- Improve the maintainability of this code and less likely to require updates
+
+### Non-Goals
+
+- Change the API to allow the user to get the support level of a single feature without retrieving the entire list.
+- Design an easier way to add a new dimension to that API.
+- Get per cluster feature-support level
+
+## Proposal
+
+- Create an interface for feature-support level that will allow us to filter the feature-support level by 
+multiple arguments. 
+- Create feature per feature ID that will implement this interface and will return the support level
+of that specific feature.
+- Change the current API to not return whole list but to return the support-level of a specific feature filtered by 
+some attributes (e.g. openshift_version, cpu_architecture, and more)
+- Get per cluster feature-support level - by given only cluster ID the service will determine the support level by 
+the cluster already exists attributes without the need to set multiple properties
+
+Implementation example:
+
+### swagger.yaml
+```swagger codegen
+  /v2/support-level/{feature_id}:
+  parameters:
+    - in: path
+      name: feature_id
+      required: true
+    - in: query
+      name: cluster_id
+    - in: query
```

Comment:
It will allow to get cluster specific feature-support-level without passing all the other arguments.
For example if the cluster OCP version is 4.12 and the CPU architecture is ARM than calling to `/v2/support-level/{feature_id}` with `cluster_id = x` will return the support-level for that feature.